src/lib/secrets/hardware_wallets.ml

```diff
@@ -0,0 +1,128 @@
+open Snark_params
+open Core
+open Signature_lib
+open Coda_base
+open Async
+
+let hardware_wallet_script = "codaledgercli"
+
+let to_bigint : string -> Bigint.t =
+ fun field ->
+  Bytes.of_string field
+  |> B58.decode Base58_check.coda_alphabet
+  |> Bytes.to_list |> List.rev |> Bytes.of_char_list |> Bytes.to_string
+  |> String.foldi ~init:Bigint.zero ~f:(fun i acc byte ->
+         Bigint.(acc lor (of_int (Char.to_int byte) lsl Int.( * ) 8 i)) )
+
+let decode_field : string -> Snark_params.Tick.Field.t =
+ fun field ->
+  to_bigint field |> Tick.Bigint.of_bignum_bigint |> Tick.Bigint.to_field
+
+let decode_scalar : string -> Snark_params.Tock.Field.t =
```

Comment:
`decode_field` and `decode_scalar` are identifical, modulo `Tick/Tock`

there could be a single function that's passed `Tick` or `Tock`